src/sentry/utils/performance_issues/performance_detection.py

```diff
@@ -353,15 +353,13 @@ def _detect_performance_problems(
     return list(unique_problems)
 
 
-def run_detector_on_data(detector, data):
-    if not detector.is_event_eligible(data):
-        return
-
-    spans = data.get("spans", [])
-    for span in spans:
-        detector.visit_span(span)
+def run_detector_on_data(detector, data, project=None):
+    if detector.is_event_eligible(data, project):
+        spans = data.get("spans", [])
+        for span in spans:
+            detector.visit_span(span)
 
-    detector.on_complete()
+        detector.on_complete()
```

Comment:
Aside from the addition of `project`, is the control flow change here just cosmetic? I think it is but just want to be sure. (FWIW I find it easier for review (and safer!) to minimize refactoring lines when you're making logic changes at the same time - basically, when possible make either logic changes or refactors but not both in a single PR 😄)

Reply:
Yes, sorry that's mb. I originally moved out `is_event_eligible` outside of `run_detector_on_data`, but moved it back. I'll revert that!

Reply:
@mjq-sentry should be good now!